cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -937,11 +1000,13 @@ def prepare_download(self, file_list, file_name, scratch_dir):
             file_list = [file_list]
 
         for ID, f in enumerate(file_list):
-            file_list[ID] = self.validated_download_file_path(scratch_dir, f)
+            file_list[ID] = self.validated_download_file_path(origin_dir, f)
+            if from_request_files_dir:
+                self.verify_access_to_file(f)
 
-        tmp_dir = tempfile.mkdtemp(prefix='download_', dir='./')
+        file_dir = tempfile.mkdtemp(prefix='download_', dir='./')
```

Comment:
The separate download directory seems to be not needed, at least when `return_archive=False`. 
(But this is a minor note, we can keep it as it is)

Reply:
it is actually used in case when `return_archive=False and file_name != file_to_download`